src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -310,18 +300,10 @@ void BaseCouplingScheme::secondExchange()
   }
 }
 
-void BaseCouplingScheme::storeExtrapolationData()
-{
-  PRECICE_TRACE(_timeWindows);
-  for (auto &data : _allData | boost::adaptors::map_values) {
-    data->storeExtrapolationData();
-  }
-}
-
 void BaseCouplingScheme::moveToNextWindow()
 {
   PRECICE_TRACE(_timeWindows);
-  for (auto &data : getAccelerationData() | boost::adaptors::map_values) {
+  for (auto &data : _allData | boost::adaptors::map_values) {
     data->moveToNextWindow();
   }
 }
```

Comment:
Regarding extrapolation this is one important bit I changed: Before, we only called `moveToNextWindow` for the accelerated data. This applied the extrapolation. Now, we move `_allData`.

Especially for explicit coupling schemes this part changed a lot: Previously this function was never called for explicit coupling schemes, because extrapolation is not reasonable for explicit coupling schemes:
```
  if (isImplicitCouplingScheme()) {
    if (not doesFirstStep()) {
      storeExtrapolationData();
      moveToNextWindow();
    }
  }
```
Now, we always call it for all coupling schemes, because all coupling schemes have to move the storage and extrapolation is just an optional feature of moving.